src/cli/index.js

```diff
@@ -398,14 +399,32 @@ export function main({
     });
 }
 
-export default function start() {
-  // ignore all arguments after a --
-  const doubleDashIndex = process.argv.findIndex(element => element === '--');
-  const startArgs = process.argv.slice(0, 2);
-  const args = process.argv.slice(2, doubleDashIndex === -1 ? process.argv.length : doubleDashIndex);
-  const endArgs = doubleDashIndex === -1 ? [] : process.argv.slice(doubleDashIndex + 1, process.argv.length);
+async function start(): Promise<void> {
+  const rc = getRcConfigForCwd(process.cwd());
+  const yarnPath = rc['yarn-path'];
 
-  main({startArgs, args, endArgs});
+  if (yarnPath && process.env.YARN_IGNORE_PATH !== '1') {
+    const argv = process.argv.slice(2);
+    const opts = {stdio: 'inherit', env: Object.assign({}, process.env, {YARN_IGNORE_PATH: 1})};
+
+    try {
+      await spawnp(yarnPath, argv, opts);
+    } catch (firstError) {
+      try {
+        await forkp(yarnPath, argv, opts);
```

Comment:
Then it will be rejected as not an executable. The alternative is to hide both errors and to throw something generic instead (like "Referenced file is not executable nor a valid javascript file").